boskos/client/client.go

```diff
@@ -0,0 +1,224 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"k8s.io/test-infra/boskos/common"
+)
+
+var (
+	defaultURL = "http://boskos"
+)
+
+// Public Boskos client object
+type Client struct {
+	url       string
+	resources []string
```

Comment:
The idea is that the client need to know what resources it's currently holding.

I lean towards to make user do update themselves? So that they can have more control, in case in some situation it does not need to update.